build.py

```diff
@@ -359,6 +359,37 @@ def get_path(self):
 
 Project.add(Tool_nuget())
 
+class Tool_perl(Tool):
+    def __init__(self):
+        Tool.__init__(self,
+            'perl',
+            archive_url = 'https://dl.hexchat.net/misc/perl/perl-5.20.0-x64.tar.xz',
+            hash = '05e01cf30bb47d3938db6169299ed49271f91c1615aeee5649174f48ff418c55')
+
+    def load_defaults(self, builder):
+        # Set the builder object to point to the path to use, when we need to pass directly
+        # the executable to *make
+        builder.perl_dir = os.path.join(builder.opts.tools_root_dir, 'perl', 'x64')
+        # full path, added to the environment when needed
+        self.perl_path = os.path.join(builder.perl_dir, 'bin')
+
+    def unpack(self):
+        # We download a tar.xz file so we estract it in the tool directory ...
+        destfile = os.path.join(self.perl_path, 'perl.exe')
+        if not os.path.isfile(destfile):
+            print_log("Unpacking perl to tools directory (%s)" % (destfile, ))
+            self.builder.make_dir(self.perl_path)
+            self.builder.exec_msys([self.builder.tar, 'xf', convert_to_msys(self.archive_file), '-C', convert_to_msys(os.path.join(self.builder.opts.tools_root_dir, 'perl'))])

```

Comment:
I thought of that, maybe expanding the tarball to handle the .zip formato based on the final part of the archive name but at the end the tools should not be too many and in this way we have a better control on what/where we install (to use the Tarball class we need to load the working dir to point to the tools dir so in the tools there will be a unpack(...), we gain a couple of lines (the makedir and the msys call).
ATM this is not necessary now but maybe later (when we change version on some tool?).